src/pushsource/_impl/model/productid.py

```diff
@@ -1,14 +1,112 @@
+from collections import defaultdict
+
+from cryptography import x509
+from frozenlist2 import frozenlist
+from pyasn1.codec.der import decoder
+
 from .base import PushItem
+from .conv import convert_maybe, sloppylist
 from .. import compat_attr as attr
 
 
+# Red Hat OID namespace is "1.3.6.1.4.1.2312.9",
+# the trailing ".1" designates a Product Certificate.
+OID_NAMESPACE = "1.3.6.1.4.1.2312.9.1."
+
+
+@attr.s()
+class Product(object):
```

Comment:
This needs to be exported from `src/pushsource/__init__.py` if it's intended as public API.